### PR TITLE
Integrate feedback from workshop session

### PR DESCRIPTION
**Pull request by eloots**

### CODE REVIEW

**Review by gianick — APPROVED** [review]

**Review by gianick — APPROVED** [review]

File: exercises/exercise_001_dotty_deprecated_syntax_rewriting/src/main/scala/org/lunatechlabs/dotty/sudoku/ReductionRules.scala [hunk 1]
```diff
@@ -4,44 +4,39 @@ object ReductionRules {
...
-        }
+    completeAndIsolatedValueSets.foldLeft(reductionSet) { case (cells, caivSet) =>
+      cells.map { cell =>
+        if (cell != caivSet) cell &~ caivSet else cell
+      }
     }
   }
 
   def reductionRuleTwo(reductionSet: ReductionSet): ReductionSet = {
     val valueOccurrences = CELLPossibleValues.map { value =>
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_001_dotty_deprecated_syntax_rewriting/src/main/scala/org/lunatechlabs/dotty/sudoku/ReductionRules.scala:19]: You're right. Will change it to Camel case

File: exercises/exercise_110_opaque_type_aliases/src/main/scala/org/lunatechlabs/dotty/sudoku/ReductionSet.scala [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  def applyReductionRuleTwo: ReductionSet = {
+    val valueOccurrences = CELLPossibleValues.map { value =>
+      cellIndexesVector.zip(reductionSet).foldLeft(Vector.empty[Int]) { case (acc, (index, cell)) =>
+        if cell contains value then index +: acc else acc
+      }
+    }
+
+    val cellIndexesToValues =
+      CELLPossibleValues.zip(valueOccurrences).groupBy((value, occurrence) => occurrence).filter { case (loc, occ) =>
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_110_opaque_type_aliases/src/main/scala/org/lunatechlabs/dotty/sudoku/ReductionSet.scala:34]: I'll apply the change to Camel case in bulk.

File: exercises/exercise_001_dotty_deprecated_syntax_rewriting/project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.11.0")
+addSbtPlugin("org.scalameta" % "sbt-mdoc" % "2.3.7")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.2")
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_001_dotty_deprecated_syntax_rewriting/project/plugins.sbt]: implemented in a4b09017a5694eaecaf48d4ef48e0fecb9b2bc05

File: exercises/exercise_101_dotty_deprecated_syntax_rewriting/project/Build.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  lazy val compileOptions = Seq(
+    "-unchecked",
+    "-deprecation",
+    "-encoding", "UTF-8",
+    "-source:future-migration",
+  )
+}
+
+object Version {
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_101_dotty_deprecated_syntax_rewriting/project/Build.scala]: Changed. See 25ec8e75d523c7de7740a51d7917c58ef08dfb53

File: exercises/exercise_102_dotty_new_control_structure_syntax/src/main/scala/org/lunatechlabs/dotty/sudoku/SudokuIO.scala [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  def sudokuPrinter(result: SudokuSolver.SudokuSolution): String =
+    result.sudoku.sliding(3, 3).map(sudokuRowPrinter).mkString("\n+---+---+---+\n", "+---+---+---+\n", "+---+---+---+")
+
+  /*
+   * FileLineTraversable code taken from "Scala in Depth" by Joshua Suereth
+   */
+
+  import java.io.{BufferedReader, File, FileReader}
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_102_dotty_new_control_structure_syntax/src/main/scala/org/lunatechlabs/dotty/sudoku/SudokuIO.scala]: Done. See 2035377110c463abfa907912dcdd9be1bbb2bc5e

File: exercises/exercise_109_union_types/project/Build.scala [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  lazy val compileOptions = Seq(
+    "-unchecked",
+    "-deprecation",
+    "-encoding", "UTF-8",
+    "-source:future-migration",
+  )
+}
+
+object Version {
+  lazy val akkaVer           = "2.6.20"
```
[the remark this answers is not in the document]
Comment (eloots) [reply exercises/exercise_109_union_types/project/Build.scala:19]: I prefer to stick with the latest version in the 2.6 release

File: exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Setting up IntelliJ or Visual Code Studio for Scala development
```
Comment (gianick) [inline exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md:1]:
```suggestion
# Setting up IntelliJ Idea or Visual Code Studio for Scala development
```

File: exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+## Setting up IntelliJ for Scala development
```
Comment (gianick) [inline exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md:3]:
```suggestion
## Setting up IntelliJ Idea for Scala development
```

File: exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+- Download the installation binary for your system (Windows, Mac, Linux) [here](https://code.visualstudio.com/download).
+
+- Follow the set-up instructions [here](https://code.visualstudio.com/docs/setup/setup-overview).
+
+- Install the Metals plugin for Visual Code Studio by following the instructions [here](https://scalameta.org/metals/docs/editors/vscode/)
+
+
+## Other Scala source editors
+
+There are plenty of other options for Source Code editing. For example, [Scala] Metals supports Vim, Sublime Text and Emacs. For more information, have a look at the [Text Editors section](https://scalameta.org/metals/docs) in the Metals Documentation
```
Comment (gianick) [inline exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md:20]: Why are you using square brackers here "[Scala] Metals"?

**Review by gianick — APPROVED** [review]

File: exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md (hunk 9 above)
Comment (eloots) [reply exercises/exercise_000_sudoku_solver_initial_state/IDE_setup.md:20]: Good question. Looking at the Metals doc, it says _"Scala language server with rich IDE features"_.

I'll change it to "Metals (Scala language server with rich IDE features)"

**Pull request merged**